substrate/frame/system/src/extensions/check_weight.rs

```diff
@@ -50,10 +50,13 @@ where
 	/// with given `DispatchClass` can have.
 	fn check_extrinsic_weight(
 		info: &DispatchInfoOf<T::RuntimeCall>,
+		len: usize,
 	) -> Result<(), TransactionValidityError> {
 		let max = T::BlockWeights::get().get(info.class).max_extrinsic;
+		let total_weight_including_length =
+			info.total_weight().add_proof_size(len.try_into().unwrap_or(u64::MAX));
```

Comment:
```suggestion
			info.total_weight().add_proof_size(len as u64);
```

Nothing more than 4GB will ever fit into the runtime.